### PR TITLE
Move samples used in Catch exception section to snippets folder

### PR DESCRIPTION
**Pull request by nemrism**
## Summary

Move samples to snippets folder from:

- samples/snippets/cpp/VS_Snippets_CLR/CatchException/CPP/catchexception1.cpp
- samples/snippets/cpp/VS_Snippets_CLR/CatchException/CPP/catchexception2cpp
- samples/snippets/csharp/VS_Snippets_CLR/CatchException/CS/catchexception1.cs
- samples/snippets/csharp/VS_Snippets_CLR/CatchException/CS/catchexception2.cs
- samples/snippets/visualbasic/VS_Snippets_CLR/CatchException/VB/catchexception1.vb
- samples/snippets/visualbasic/VS_Snippets_CLR/CatchException/VB/catchexception2.vb

Contributes to #15006

### Internal preview

- [How to use the try/catch block to catch exceptions](https://review.docs.microsoft.com/en-us/dotnet/standard/exceptions/how-to-use-the-try-catch-block-to-catch-exceptions?branch=pr-en-us-24084)
- [How to use specific exceptions in a catch block](https://review.docs.microsoft.com/en-us/dotnet/standard/exceptions/how-to-use-specific-exceptions-in-a-catch-block?branch=pr-en-us-24084)

### CODE REVIEW

File: docs/standard/exceptions/snippets/how-to-use-specific-exceptions-in-a-catch-block/csharp/catchexception.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public static void PromoteEmployee(Object emp)
         e.Emlevel = e.Emlevel + 1;
     }
 
-    public static void Main()
+    static void Main(string[] args)
```
Comment (IEvangelist) [inline docs/standard/exceptions/snippets/how-to-use-specific-exceptions-in-a-catch-block/csharp/catchexception.cs:32]:
```suggestion
    static void Main()
```

No reason to add `args` if they're not being used. It's perfectly valid to have a [`void Main()`](https://docs.microsoft.com/dotnet/csharp/programming-guide/main-and-command-args/#overview).

**Comment by IEvangelist** [comment]
Hey, @adegeo and @BillWagner is Snippets 5000 designed to compile C++, I do not recall whether or not we've been handling that?

File: docs/standard/exceptions/how-to-use-the-try-catch-block-to-catch-exceptions.md [hunk 2]
```diff
@@ -18,8 +19,9 @@ Place any code statements that might raise or throw an exception in a `try` bloc
 
 In the following example, a <xref:System.IO.StreamReader> opens a file called *data.txt* and retrieves a line from the file. Since the code might throw any of three exceptions, it's placed in a `try` block. Three `catch` blocks catch the exceptions and handle them by displaying the results to the console.
 
-[!code-csharp[CatchException#3](~/samples/snippets/csharp/VS_Snippets_CLR/CatchException/CS/catchexception2.cs#3)]
-[!code-vb[CatchException#3](~/samples/snippets/visualbasic/VS_Snippets_CLR/CatchException/VB/catchexception2.vb#3)]
+:::code language="cpp" source="./snippets/how-to-use-the-try-catch-block-to-catch-exceptions/cpp/catchexception.cpp" id="Snippet3":::
```
Comment (IEvangelist) [inline docs/standard/exceptions/how-to-use-the-try-catch-block-to-catch-exceptions.md:22]: Thank you for adding C++ here, much appreciated.

**Comment by BillWagner** [comment]
ping @adegeo 

The build failure here is (I think) the same one we fixed in docs-desktop yesterday. Would you agree? If so, I'll open a PR for this repo and get it merged.

File: docs/standard/exceptions/how-to-use-the-try-catch-block-to-catch-exceptions.md (hunk 2 above)
Comment (nemrism) [reply docs/standard/exceptions/how-to-use-the-try-catch-block-to-catch-exceptions.md:22]: I've seen it while preparing the move of the other page samples.
I also left some unused files in the samples folder, I did not find any usage of them, but did not want to delete them anyway.

**Comment by adegeo** [comment]
The error here is that there isn't a project for the CPP code files.

**Review by BillWagner — APPROVED** [review]
Approving and merging. (The last comment on the CI build got lost while out of office).

I apologize for the delay.

**Pull request merged**